screens/LoginSignUp.js

```diff
@@ -0,0 +1,236 @@
+import React, {useState, useRef, useEffect} from 'react';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+import {
+  StyleSheet,
+  Text,
+  View,
+  Image,
+  TouchableOpacity,
+  TouchableHighlight,
+  TextInput,
+  Linking,
+  Alert,
+} from 'react-native';
+import auth from '@react-native-firebase/auth';
+
+const LoginSignUp = props => {
+  const [isLogin, setVisible] = useState(true);
+  const inputPasswordRef = useRef(null);
+  const [user, setUser] = useState(null);
+  const [email, setEmail] = useState(null);
+  const [password, setPassword] = useState(null);
+
+  function isStateChanged(user) {
+    setUser(user);
+  }
+
+  useEffect(() => {
+    const subscriber = auth().onAuthStateChanged(isStateChanged);
+
+    return subscriber;
+  }, []);
+
+  const handleLogin = () => {
+    if (isLogin) {
+      auth()
+        .signInWithEmailAndPassword(email, password)
+        .then(result => {
+          goToMainScreen();
+        })
+        .catch(error => {
+          Alert.alert('Wrong credentials, please check email or password!');
+        });
+    } else {
+      auth()
+        .createUserWithEmailAndPassword(email, password)
+        .then(data => {
+          goToMainScreen();
+        })
+        .catch(error => {
+          if (error.code === 'auth/email-already-in-use') {
+            Alert.alert('That email address is already in use!');
+          }
+
+          if (error.code === 'auth/invalid-email') {
+            Alert.alert('That email address is invalid!');
+          }
+        });
+    }
+  };
+
+  return (
+    <KeyboardAwareScrollView
+      contentContainerStyle={{flexGrow: 1, backgroundColor: '#F2F2F2'}}
+      style={styles.removePadding}>
+      <View style={[styles.alignImage, styles.whiteBackground]}>
+        <Image source={require('../assets/images/loginImage.png')} />
+      </View>
+
+      <View style={[styles.tabContainer, styles.whiteBackground]}>
+        <TouchableHighlight
+          onPress={() => {
+            setVisible(true);
+          }}
+          underlayColor="white"
+          style={styles.click}>
+          <Text style={[styles.buttonLeft]}>Login</Text>
+        </TouchableHighlight>
+        <TouchableHighlight
+          onPress={() => {
+            setVisible(false);
+          }}
+          underlayColor="white"
+          style={styles.click}>
+          <Text style={[styles.buttonRight, styles.textCenter]}>Signup</Text>
+        </TouchableHighlight>
+      </View>
+
+      {isLogin ? (
+        <View style={styles.active} />
+      ) : (
+        <View style={[styles.active, styles.reverse]} />
+      )}
+
+      <Text style={[styles.inputLabel, styles.spacer]}>Email address</Text>
+      <TextInput
+        style={styles.input}
+        onSubmitEditing={() => {
+          inputPasswordRef.current.focus();
```

Comment:
Safely access these cases like 
```suggestion
          inputPasswordRef?.current?.focus();
```